### ISSUE TITLE
Identifier getters in proxies are non-lazy if identifier properties are in traits

### ISSUE THREAD

**Issue by amcsi** [doctrine/common#800]
When generating proxies from documents: if a document has its `Doctrine\ODM\MongoDB\Mapping\Annotations\Id` annotation set on the `$id` property it is getting from a trait, the proxy gets generated and works, but in the proxy class the `getId()` field is non-lazy like any other property like so:

```php
/**
 * {@inheritDoc}
 */
public function getId()
{

    $this->__initializer__ && $this->__initializer__->__invoke($this, 'getId', []);

    return parent::getId();
}
```

If I move the `$id` property to the document either directly or to any of its ancestor classes indirectly, then in the generated proxy the `getId()` method gets generated to be lazy as expected:

```php
/**
 * {@inheritDoc}
 */
public function getId()
{
    if ($this->__isInitialized__ === false) {
        return  parent::getId();
    }


    $this->__initializer__ && $this->__initializer__->__invoke($this, 'getId', []);

    return parent::getId();
}
```

In short: IDs in *traits* are non-lazy

**Comment by Ocramius** [comment]
Duplicate of https://github.com/doctrine/common/issues/587
Duplicate of https://github.com/doctrine/common/pull/356
